.github/workflows/unit-tests.yml

```diff
@@ -112,8 +122,41 @@ jobs:
 
       - name: save warning report
         if: success()
-        uses: actions/upload-artifact@v2
+        uses: actions/upload-artifact@v3
         with:
           name: pytest-warning-report-html
           path: |
             reports/pytest_warnings/warning_report_all.html
+
+  # Combine and upload coverage reports.
+  coverage:
+    needs: run-tests
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [ 3.8 ]
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v3
+
+      - name: Setup Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Download all artifacts
+        uses: actions/download-artifact@v3
+        with:
+          name: coverage
+          path: reports
+
+      - name: Install Python dependencies
+        run: |
+          pip install -r requirements/edx/coverage.txt
+
+      - name: Run coverage
+        run: |
+          coverage combine reports/*
```

Comment:
if we run `coverage combine` it does not get data files. that's why we have to explicitly pass path `reports/*` as an argument

_**Failed Run**_: https://github.com/openedx/edx-platform/actions/runs/4052767700/jobs/6972895452